### PR TITLE
Log4j upgrade for Security Vulnerability CVE-2021-44832

### PR DESCRIPTION
**Pull request by vvitvits**
This PR upgrades log4j to v 2.3.2 in order to fix issue #133. This upgrade introduces breaking changes related to log4j configuration. See readme for more details.

### CODE REVIEW

File: src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="info" strict="true" name="XMLConfigTest"
+               packages="org.apache.logging.log4j.test">
```
Comment (vvitvits) [inline src/main/resources/log4j2.xml:3]: @mhemker is this packages here needed? I think its breaking logging for main app.

Comment (mhemker-sfmc) [reply src/main/resources/log4j2.xml:3]: @vvitvits I don't think you need that entire attribute.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,10 @@ Java platform. Among other things, the SDK:
 For more information about the Java SDK and how to use it, please see
 the Javadocs at http://salesforce-marketingcloud.github.io/FuelSDK-Java/.
 
+New Features in Version 1.6.0
+------------
+* This version upgrades SDK to use log4j version 2.3.2 that contains fix for [Security Vulnerability CVE-2021-44832](https://logging.apache.org/log4j/2.x/security.html#CVE-2021-44832). Log4j upgrade introduces breaking changes to the way log4j is configured. This version of SDK is using Log4j2 bridge to help with version migration. If client overrides log4j properties they might need to be converted to the new log4j2 format. Please see this link for more details on migrating to log4j v2: https://logging.apache.org/log4j/log4j-2.3.2/manual/migration.html.
```
Comment (mhemker-sfmc) [inline README.md:30]: You should mention that a sample log4j2.xml is provided.

**Review by manivinesh — APPROVED** [review]
Looks Good to Me. I believe we ran the test and made sure everything looks fine.

**Review by mhemker-sfmc — APPROVED** [review]

**Pull request merged**